src/net/sourceforge/kolmafia/textui/command/ClosetCommand.java

```diff
@@ -57,35 +59,28 @@ public void run(final String cmd, String parameters) {
       return;
     }
 
-    int meatAttachmentCount = 0;
-    long meatCount = 0;
-    int hatCount = 0;
+    var split = Arrays.stream(itemList).collect(Collectors.partitioningBy(AdventureResult::isMeat));
+    var meat = split.get(true);
+    var items = split.get(false);
 
-    for (int i = 0; i < itemList.length; ++i) {
-      AdventureResult item = itemList[i];
-      if (item.getName().equals(AdventureResult.MEAT)) {
-        meatCount += item.getLongCount();
-        meatAttachmentCount += 1;
-        itemList[i] = null;
-      } else if (EquipmentDatabase.isHat(item)) {
-        hatCount += 1;
+    if (meat.size() > 0) {
+      int meatCount =
+          meat.stream().map(AdventureResult::getCount).mapToInt(Integer::intValue).sum();
+      if (meatCount > 0) {
```

Comment:
Will this still work if the object is an AdventureLongCountResult?  Line 67 in the deleted code handled that but I not comfortable enough with streams to be certain the new line 68 will.  I will consider this blocking until I am told I am wrong :-)

Reply:
good spot, I think this one should also be `getLongCount`. I will add a test.

Reply:
Fixed, and also had to adjust `ItemFinder`, which was parsing the number as an int. I guess we didn't spot this before because it's extremely rare, and the previous behaviour went 3 billion -> 0 -> all meat, which was probably close to what people wanted out of this.

Reply:
IIRC we found characters with meat quantities that overflowed an int.  The fix was AdventureLongCountResult.  There was a deliberate decision not to just make the count long in AdventureResult and a deliberate decision not to change every reference to a long, only the ones we expected might someday overflow.  Since ItemFinder can parse meat that was probably missed.